ethereum/core/src/test/java/tech/pegasys/teku/core/signatures/ValidatorSigningRecordTest.java

```diff
@@ -0,0 +1,150 @@
+    assertThat(result).isEqualToComparingFieldByField(record);
+  }
+
+  @ParameterizedTest(name = "signBlock({0})")
+  @MethodSource("blockCases")
+  void signBlock(
+      @SuppressWarnings("unused") final String name,
+      final ValidatorSigningRecord input,
+      final UnsignedLong slot,
+      final Optional<ValidatorSigningRecord> expectedResult) {
+    assertThat(input.maySignBlock(slot)).isEqualTo(expectedResult);
+  }
+
+  static List<Arguments> blockCases() {
+    final ValidatorSigningRecord startingRecord =
+        new ValidatorSigningRecord(
+            UnsignedLong.valueOf(3), UnsignedLong.valueOf(6), UnsignedLong.valueOf(7));
+    return List.of(
+        Arguments.of(
+            "noExistingRecord",
+            new ValidatorSigningRecord(),
+            UnsignedLong.ONE,
+            Optional.of(
+                new ValidatorSigningRecord(
+                    UnsignedLong.ONE,
+                    ValidatorSigningRecord.NEVER_SIGNED,
+                    ValidatorSigningRecord.NEVER_SIGNED))),
+        Arguments.of("=", startingRecord, UnsignedLong.valueOf(3), Optional.empty()),
+        Arguments.of("<", startingRecord, UnsignedLong.valueOf(2), Optional.empty()),
+        Arguments.of(
+            ">",
+            startingRecord,
+            UnsignedLong.valueOf(4),
+            Optional.of(
+                new ValidatorSigningRecord(
+                    UnsignedLong.valueOf(4), UnsignedLong.valueOf(6), UnsignedLong.valueOf(7)))));
+  }
+
+  @ParameterizedTest(name = "maySignAttestation({0})")
+  @MethodSource("attestationCases")
+  void maySignAttestation(
+      @SuppressWarnings("unused") final String name,
+      final ValidatorSigningRecord input,
+      final UnsignedLong sourceEpoch,
+      final UnsignedLong targetEpoch,
+      final Optional<ValidatorSigningRecord> expectedResult) {
+    assertThat(input.maySignAttestation(sourceEpoch, targetEpoch)).isEqualTo(expectedResult);
+  }
+
+  static List<Arguments> attestationCases() {
+    final ValidatorSigningRecord startingRecord =
+        new ValidatorSigningRecord(
+            UnsignedLong.ONE, UnsignedLong.valueOf(4), UnsignedLong.valueOf(6));
+    return List.of(
+        // No record
+        attestationArguments(
+            "NEVER_SIGNED",
+            "NEVER_SIGNED",
+            new ValidatorSigningRecord(),
+            1,
+            2,
+            Optional.of(
+                new ValidatorSigningRecord(
+                    UnsignedLong.ZERO, UnsignedLong.valueOf(1), UnsignedLong.valueOf(2)))),
+        attestationArguments("=", "=", startingRecord, 4, 6, Optional.empty()),
+        attestationArguments("=", "<", startingRecord, 4, 5, Optional.empty()),
+        attestationArguments(
+            "=",
+            ">",
+            startingRecord,
+            4,
+            7,
+            Optional.of(
+                new ValidatorSigningRecord(
+                    UnsignedLong.ONE, UnsignedLong.valueOf(4), UnsignedLong.valueOf(7)))),
```

Comment:
potentially if the ValidatorSigningRecord declaration is pulled out so this can collapse to one line, then it'd be a lot more readable.
spotless wrapping it in this way really makes it difficult to read.

Reply:
Found some utility methods I could extract to make it work better with spotless.